encoding/protobuf.go

```diff
@@ -112,19 +113,42 @@ func (p protoSerializer) Encoding() Encoding {
 	return Protobuf
 }
 
+func (p protoSerializer) IsClientStreaming() bool {
+	return p.method.IsClientStreaming()
+}
+
+func (p protoSerializer) IsServerStreaming() bool {
+	return p.method.IsServerStreaming()
+}
+
+func (p protoSerializer) StreamRequest() ([]byte, error) {
```

Comment:
The name of this API is odd.

We have `Request()` and `StreamRequest()`, one is returning a `*transport.Request` and the other one returns a `[]byte`. We should rename this method IMO, maybe `NextStreamRequestBody()`